sigopt_example.m

```diff
@@ -1,40 +1,51 @@
-% 
-%
-% define dimensions
-%
-% make connection object
-conn = py.sigopt_matlab.get_connection()
+import sigopt.Connection
 
-% dimension 1
-x_bounds = py.dict(pyargs('min', 0, 'max', 1))
-x_parameter = py.dict(pyargs('name', 'x', ...
-                             'type', 'double', ...
-                             'bounds', x_bounds))
-                        
-% dimension 2                
-y_bounds = py.dict(pyargs('min', 0, 'max', 1))
-y_parameter = py.dict(pyargs('name', 'y', ...
-                             'type', 'double', ...
-                             'bounds', y_bounds))
+conn = Connection('<YOUR API TOKEN HERE>')
 
-                    
-% define parameters and name variables
-parameters = py.list({x_parameter, y_parameter})
-name = 'Franke Optimization (MATLAB)'
+experiment = conn.experiments().create(struct( ...
+  'name', 'Franke Optimization (MATLAB)', ...
+  'parameters', [ ...
+    struct( ...
+      'name', 'x', ...
+      'type', 'double', ...
+      'bounds', struct( ...
+        'min', 0, ...
+        'max', 1 ...
+      ) ...
+    ), ...
+    struct( ...
+      'name', 'y', ...
+      'type', 'double', ...
+      'bounds', struct( ...
+        'min', 0, ...
+        'max', 1 ...
+      ) ...
+    ) ...
+  ] ...
+))
 
-% make experiment
-experiment = py.sigopt_matlab.make_new_experiment(name, parameters, conn)
-experiment_id = py.getattr(experiment, 'id')
+experiment_id = experiment.id;
 
 % optimize!
-num_iterations = 30
+num_iterations = 30;
 for n = [1:num_iterations]
-    % get suggestion assignments (point in the domain)
-    suggestion = py.sigopt_matlab.get_new_suggestion(experiment_id, conn)
-    suggestion_id = suggestion{1}
-    assignments = suggestion{2}
-    % plug in the point into your function/simulation
-    value = franke(assignments{'x'}, assignments{'y'})
-    % report observed value
-    py.sigopt_matlab.report_observation(experiment_id, suggestion_id, value, conn)
-end
\ No newline at end of file
+  % get suggestion assignments (point in the domain)
+  suggestion = conn.experiments(experiment_id).suggestions().create()
+  suggestion_id = suggestion.id;
+  assignments = suggestion.assignments;
+  % plug in the point into your function/simulation
+  value = franke(assignments.x, assignments.y)
+  % report observed value
+  observation = conn.experiments(experiment_id).observations().create(struct( ...
+    'suggestion', suggestion_id, ...
+    'value', value ...
+  ))
+end
+
+% Example function - replace this with the function you are trying to optimize
+function value = franke(x,y)
```

Comment:
This is interesting, you did not used to be able to define a function within a script.  You could define multiple functions in a file but not within a script that could be called.